skimage/transform/registration.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+from scipy.optimize import minimize
+from scipy import ndimage as ndi
+
+from .pyramids import pyramid_gaussian
+from ..measure import compare_mse
+
+__all__ = ['register_affine']
+
+
+def _argmin_to_matrix(argmin):
+    """
+    Converts the optimisation parameters to a 3x3 transformation matrix
+
+    The optimisation paramters are known as the argmin and are composed of
+    the first 2 rows of the transformation matrix, as that is all that
+    is used in an affine transformation.
+
+    Parameters
+    ----------
+    argmin : (6) array
+        Input array giving the argument of the minimum function to optimise against
+
+    Returns
+    -------
+    matrix : (3, 3) array
+        A transformation matrix used to obtain a new image
+    """
+
+    matrix = np.empty((3, 3))
+
+    matrix[0][0] = argmin[0]
+    matrix[0][1] = argmin[1]
+    matrix[0][2] = argmin[2]
+    matrix[1][0] = argmin[3]
+    matrix[1][1] = argmin[4]
+    matrix[1][2] = argmin[5]
+    matrix[2][0] = 0
+    matrix[2][1] = 0
+    matrix[2][2] = 1
+
+    return matrix
+
+
+def _matrix_to_argmin(matrix):
+    """
+    Converts a 3x3 transformation matrix to the optimisation parameters
+
+    See the inverse function `_argmin_to_matrix`.
+
+    Parameters
+    ----------
+    matrix : (3, 3) array
+        A transformation matrix used to obtain a new image
+
+    Returns
+    -------
+    argmin : (6) array
+        Output array giving the argument of the minimum function to optimise against
+
+    """
+
+    return (matrix[0][0], matrix[0][1], matrix[0][2], matrix[1][0], matrix[1][1], matrix[1][2])
+
+
+def register_affine(reference, target, *, cost=compare_mse, nlevels=None,
+                    iter_callback=lambda img, matrix: None):
+    """
+    Returns a matrix which registers the target image to the reference image
+
+    Uses image registration techniques, and only optimises with affine transformations.
```

Comment:
Be *really* specific about what "techniques" means here, or reword the sentence if you mean to say it's image registration using affine transformations.